tests/Integration/Behaviour/Features/Scenario/Discount/FO/full_user_experience_test_with_discount_free_shipping.feature

```diff
@@ -34,7 +37,44 @@ Feature: Full UX discount test
     And I use a voucher "complete_free_shipping_discount" on the cart "dummy_cart"
     And cart "dummy_cart" total with tax included should be '$19.81'
     Then my cart "dummy_cart" should have the following details:
-      | total_products           | $19.81 |
-      | shipping                 | $7.00  |
-      | total_discount           | -$7.00 |
-      | total                    | $19.81 |
+      | total_products | $19.81 |
+      | shipping       | $7.00  |
+      | total_discount | -$7.00 |
+      | total          | $19.81 |
+
+  Scenario: Create and update a complete discount with free shipping using new CQRS
+    Given I create an empty cart "dummy_cart" for customer "testCustomer"
+    And there is a product in the catalog named "product1" with a price of 19.812 and 1000 items in stock
+    When I create a "free_shipping" discount "complete_free_shipping_discount" with following properties:
+      | name[en-US] | Promotion           |
+      | name[fr-FR] | Promotion_fr        |
+      | active      | true                |
+      | valid_from  | 2025-01-01 11:05:00 |
+      | valid_to    | 2025-12-01 00:00:00 |
+      | code        | PROMO_2025_2        |
+    And discount "complete_free_shipping_discount" should have the following properties:
+      | name[en-US] | Promotion           |
+      | name[fr-FR] | Promotion_fr        |
+      | active      | true                |
+      | valid_from  | 2025-01-01 11:05:00 |
+      | valid_to    | 2025-12-01 00:00:00 |
+      | code        | PROMO_2025_2        |
+    Then I update discount "complete_free_shipping_discount" with the following properties:
+      | name[en-US] | Promotion_2    |
+      | name[fr-FR] | Promotion_fr_2 |
+    And discount "complete_free_shipping_discount" should have the following properties:
+      | name[en-US] | Promotion_2         |
+      | name[fr-FR] | Promotion_fr_2      |
+      | active      | true                |
+      | valid_from  | 2025-01-01 11:05:00 |
+      | valid_to    | 2025-12-01 00:00:00 |
+      | code        | PROMO_2025_2        |
+    And I add 1 product "product1" to the cart "dummy_cart"
+    And cart "dummy_cart" total with tax included should be '$26.81'
+    And I use a voucher "complete_free_shipping_discount" on the cart "dummy_cart"
+    And cart "dummy_cart" total with tax included should be '$19.81'
+    Then my cart "dummy_cart" should have the following details:
+      | total_products | $19.81 |
+      | shipping       | $7.00  |
+      | total_discount | -$7.00 |
+      | total          | $19.81 |
```

Comment:
You can remove this scenario it doesn't bring more value

The BO scenarios should handle the tests for update and ensure the discount is correctly saved and persisted in database The FO scenarios ensure that given a discount with a specific state we have the appropriate behaviour But checking that we create, then edit, then get the correct behaviour is redundant Especially since here the only thing modified is the name which has no impact on what we test afterward

Reply:
```suggestion
```